### PR TITLE
Add EdgeSwitch 10XP definition

### PR DESCRIPTION
**Pull request by computman007**
.1.3.6.1.2.1.1.1.0 = STRING: "RTL8380-8GE-2GEF"
.1.3.6.1.2.1.1.2.0 = OID: .1.3.6.1.4.1.27282.3.2.10

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by sorano** [comment]
@computman007 

Please add test data so we can ensure your change is not broken in the future.

Read the docs to find out how: https://docs.librenms.org/Developing/os/Test-Units

**Comment by computman007** [comment]
```
computman@vlibrenms:/opt/librenms$ ./scripts/pre-commit.php --db --snmpsim
Only checking os: edgeswitch
Running unit tests...
Starting snmpsim listening on 127.1.6.2:1162...

In bootstrap.php line 62:

  SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'


Style check skipped.
Lint check skipped.
Tests ok, submit away :)
computman@vlibrenms:/opt/librenms$

```

I don't know if that enought

[edgeswitch.zip](https://github.com/librenms/librenms/files/3931809/edgeswitch.zip)

**Comment by PipoCanaja** [comment]
Hi @computman007 
The json file would be the 2nd half of the test material. The 1st half is the snmp capture (.snmprec) file. 

The idea is to capture the snmp values, and also record in the json how these snmp values are interpreted. That builds some sort of "picture" of the device. Then, later, when somebody patches LibreNMS, all those snmp files are processed again and the result is compared to the saved picture. This ensures that we don't have regressions. 

Basically, we need you to : 
- generate one snmprec file, with the "-v 10xp" (variant 10xp) option enabled (so that it does not collide with existing test-data). To do this, you have to run the ./scripts/collect-snmp-data.php -h <deviceid> -v 10xp 
- and add the resulting file to the PR. 
- generate the matching json file (./scripts/save-test-data.php -o edgeswitch -v 10xp). If that step fails (the config for this is not that easy currently) I can do it.

**Comment by computman007** [comment]
> Hi @computman007
> The json file would be the 2nd half of the test material. The 1st half is the snmp capture (.snmprec) file.
> 
> The idea is to capture the snmp values, and also record in the json how these snmp values are interpreted. That builds some sort of "picture" of the device. Then, later, when somebody patches LibreNMS, all those snmp files are processed again and the result is compared to the saved picture. This ensures that we don't have regressions.
> 
> Basically, we need you to :
> 
> * generate one snmprec file, with the "-v 10xp" (variant 10xp) option enabled (so that it does not collide with existing test-data). To do this, you have to run the ./scripts/collect-snmp-data.php -h  -v 10xp
> * and add the resulting file to the PR.
> * generate the matching json file (./scripts/save-test-data.php -o edgeswitch -v 10xp). If that step fails (the config for this is not that easy currently) I can do it.

The thing is that I'm not a dev and some part actions are difficult to understand.

I have modified my yaml file locally (it works) and I can run scripts for you. I just need the different script to run in which order and tell me the file I have to send to you :)

**Comment by PipoCanaja** [comment]
That's exactly what I did yesterday:) 

- ```./scripts/collect-snmp-data.php -h {deviceID} -v 10xp```
- and add the resulting file to the PR.
- ```./scripts/save-test-data.php -o edgeswitch -v 10xp ```
- and add the resulting file to the PR.

**Comment by computman007** [comment]
> * ./scripts/collect-snmp-data.php -h {deviceID} -v 10xp

First command answer : 
`Updated snmprec data /opt/librenms/tests/snmpsim/edgeswitch_10xp.snmprec`

Second command answer :
```
computman@vlibrenms:/opt/librenms$ ./scripts/save-test-data.php -o edgeswitch -v 10xp
Starting snmpsim listening on 127.1.6.1:1161...
OS: edgeswitch
Module: all
Variant: 10xp


In ModuleTestHelper.php line 504:

  /opt/librenms/tests/snmpsim/edgeswitch_10xp.snmprec does not exist!
```

**Comment by computman007** [comment]
> That's exactly what I did yesterday:)
> 
> * `./scripts/collect-snmp-data.php -h {deviceID} -v 10xp`
> * and add the resulting file to the PR.
> * `./scripts/save-test-data.php -o edgeswitch -v 10xp `
> * and add the resulting file to the PR.

First command need to be executed as root to generate the file .snmprec but OK

Second command show errors

As root 
ERROR: cant drop priveleges: Must drop priveleges to a non-priveleged user&group

As normal user
```
In Connection.php line 664:

  SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost' (SQL: select * from `devices` where `hostname` = 127.1.6.1 limit 1)


In PDOConnection.php line 31:

  SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'


In PDOConnection.php line 27:

  SQLSTATE[HY000] [1698] Access denied for user 'root'@'localhost'
```

**Comment by PipoCanaja** [comment]
Please attach the snmprec file, I'll take care of the 2nd step, as this one is a little bit more complicated to setup. 

The 1st command should succeed as librenms user. You should never use "root" account anywhere as it creates issues for updating files. (probably what's happening to you right now. )

Try to run ./validate.php to verify and fix if necessary your permissions. 

Bye

**Comment by computman007** [comment]
> Please attach the snmprec file, I'll take care of the 2nd step, as this one is a little bit more complicated to setup.
> 
> The 1st command should succeed as librenms user. You should never use "root" account anywhere as it creates issues for updating files. (probably what's happening to you right now. )
> 
> Try to run ./validate.php to verify and fix if necessary your permissions.
> 
> Bye

[edgeswitch_10xp.zip](https://github.com/librenms/librenms/files/3935770/edgeswitch_10xp.zip)

**Comment by PipoCanaja** [comment]
Hi @computman007 
Looks very good. It is ready for merge now. Thanx for your contribution ! 
Bye

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-59-release-changelog-december-2019/10558/1
